### PR TITLE
Add prefetch support for aggregative nested loop join

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

**Comment by mustafasrepo** [comment]
@ozankabak This PR is ready for your review. By the way it still fails from MSRV test. I will fix this failure.

File: datafusion/physical-plan/src/joins/sliding_window_join_utils.rs [hunk 1]
```diff
@@ -1420,6 +1421,326 @@ impl DerefMut for BroadcastStreams {
...
+
+/// Retrieves the number of rows to that is guaranteed to match with future probe values
+/// based on certain matching criteria.
+///
+/// The function takes a reference to a `RecordBatch`, a `JoinFilter`, and a slice of tuples
+/// containing `PhysicalSortExpr` and `Interval`. It evaluates the filter conditions and performs
+/// a binary search to determine the length of the record batch that satisfies the filter condition,
+/// considering the specified sort expressions and intervals.
+///
+/// # Arguments
```
Comment (ozankabak) [inline datafusion/physical-plan/src/joins/sliding_window_join_utils.rs:1679]:
```suggestion
/// # Parameters
```

**Review by ozankabak — COMMENTED** [review]
Let's address the first wave of reviews, and I will review `join_pipeline_selection.rs` after rebase.

File: datafusion/physical-plan/src/joins/sliding_window_join_utils.rs [hunk 2]
```diff
@@ -1420,6 +1421,326 @@ impl DerefMut for BroadcastStreams {
...
+    Start(usize),
+    /// Fetch data from the end
+    End(usize),
+    /// Fetch data both from the start and from the end
+    Both((usize, usize)),
+}
+
+impl FetchMode {
+    /// Constructs a new `FetchBuffer` given the fetch sizes from the start and end.
+    pub fn new_from_vals(fetch_start: usize, fetch_end: usize) -> Self {
```
[the remark this answers is not in the document]
Comment (mustafasrepo) [reply datafusion/physical-plan/src/joins/sliding_window_join_utils.rs:1437]: Changed it to new

**Comment by mustafasrepo** [comment]
I have addressed all of the reviews, it should be ready to merge after rebase completed.

**Review by ozankabak — APPROVED** [review]
LGTM, ready for merge after CI passes.

**Pull request merged**